src/material-experimental/dialog/dialog-config.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {ViewContainerRef} from '@angular/core';
+import {Direction} from '@angular/cdk/bidi';
+import {ComponentType} from '@angular/cdk/overlay';
+import {CdkDialogContainer} from './dialog-container';
+
+/** Valid ARIA roles for a dialog element. */
+export type DialogRole = 'dialog' | 'alertdialog';
+
+/** Possible overrides for a dialog's position. */
+export interface DialogPosition {
+  top?: string;
+  bottom?: string;
+  left?: string;
+  right?: string;
+}
+
+export class DialogConfig<D = any> {
```

Comment:
Should all the symbols be prefixed with `Cdk`? E.g. `CdkDialogConfig`.

Reply:
@jelbourn said that he wanted to have `Cdk` prefix only on Directives within the CDK.